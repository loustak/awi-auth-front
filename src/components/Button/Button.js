import React from 'react'
import styles from './Button.module.css'
import { Button as BButton } from 'react-bootstrap'

function Button(props) {
  return (
    <BButton
      className={styles.button}
      variant={props.variant || 'blue'}
      style={{ borderRadius: (props.shape === 'round' ? '50px' : props.shape === 'square' ? '0' : '0.25rem') }}
      {...props}
    >
      {props.children}
    </BButton>
  )
}

export default Button
