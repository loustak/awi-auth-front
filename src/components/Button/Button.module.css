.button {
    padding-left: 20px;
    padding-right: 20px;
    /*min-width: 120px;*/
}
