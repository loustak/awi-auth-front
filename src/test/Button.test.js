import React from 'react'
import { create } from 'react-test-renderer'
import Button from '../components/Button/Button'

describe('Button component test', () => {
  test('Matches the snapshot', () => {
    const button = create(<Button />)
    expect(button.toJSON()).toMatchSnapshot()
  })
})
