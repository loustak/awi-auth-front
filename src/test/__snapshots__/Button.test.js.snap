// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test Matches the snapshot 1`] = `
<button
  className="button btn btn-blue"
  disabled={false}
  style={
    Object {
      "borderRadius": "0.25rem",
    }
  }
  type="button"
/>
`;
